packages/flame/lib/src/sprite_batch.dart

```diff
@@ -224,13 +276,15 @@ class SpriteBatch {
       color: color ?? defaultColor,
     );
 
+    if (flip && useAtlas) _needsFlips = true;
```

Comment:
If you replace `_needsFlips` variable with `_hasFlips`, then this could simply be
```
if (flip && useAtlas && !_hasFlips) {
  _hasFlips = true;
  _generateFlippedAtlas();
}
```
and there's no need for separate `atlas` and `flippedAtlas` variables.

Reply:
The only thing missing from this is caching the flipped atlas - this is what I've been stuck on and circling around as well. I think the caching is still important because every layer of a map that is loaded will reload the same set of atlas images each time. So at a minimum caching prevents regenerating for each layer of a map.

The suggestions you made here and @spydon comment as well has given me an idea on how to get the caching working though

Reply:
Since you're using `fetchOrGenerate` for the flipped atlas, isn't it already being cached?